drivers/disk/usdhc.c

```diff
@@ -2635,10 +2683,23 @@ static int usdhc_board_access_init(struct usdhc_priv *priv)
 
 	if (!priv->detect_gpio) {
 		LOG_INF("USDHC detection other than GPIO");
-		/* DATA3 does not monitor card insertion */
-		base->PROT_CTRL &= ~USDHC_PROT_CTRL_D3CD_MASK;
+		if (config->detect_dat3) {
+			base->PROT_CTRL |= USDHC_PROT_CTRL_D3CD_MASK;
+			/* Pull down DAT3 */
+			usdhc_dat3_pull(USDHC_DAT3_PULL_DOWN, priv);
+			/* Delay to allow pin state to change */
+			usdhc_millsec_delay(100);
```

Comment:
magic number... do we have some reference on what validates this?

Reply:
The SDK has a delay placed there of 1 ms, I'm not sure where I pulled 100ms from. I've removed the delay and DAT3 card detection still works fine.

Reply:
not necessarily requesting to remove it... just wanted to understand where it was coming from. I have a personal bias against magic numbers. Just find it useful to have comments/references to validate their use for a future reader of the code.